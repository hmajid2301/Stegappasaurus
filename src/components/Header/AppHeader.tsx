import React from 'react';
import {TouchableOpacity} from 'react-native';
import {Icon} from 'react-native-elements';
import {NavigationScreenProp} from 'react-navigation';

import {ITheme, TabColors} from '~/constants/types';
import Header from './Header';

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  primary: TabColors;
  theme: ITheme;
}

export default class AppHeader extends React.Component<IProps, {}> {
  public render() {
    return (
      <Header
        {...this.props}
        left={this.leftComponent()}
        styles={{height: 50}}
      />
    );
  }

  private leftComponent = () => (
    <TouchableOpacity onPress={this.toHome} testID="home">
      <Icon
        color={this.props.theme.color}
        name="arrow-back"
        type="material-icons"
      />
    </TouchableOpacity>
  );

  private toHome = () => {
    this.props.navigation.navigate('Main');
  };
}
