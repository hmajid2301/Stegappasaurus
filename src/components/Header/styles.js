import { StyleSheet } from 'react-native';


const styles = StyleSheet.create({
  container: {
    paddingTop: 20,
    height: 70,
    borderBottomWidth: 2,
  },
});

export default styles;
