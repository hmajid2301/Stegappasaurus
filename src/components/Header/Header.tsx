import React from 'react';
import {StatusBar, StyleProp, View, ViewStyle} from 'react-native';
import {
  Header as ElementsHeader,
  HeaderSubComponent,
} from 'react-native-elements';
import {NavigationScreenProp} from 'react-navigation';

import {ITheme, TabColors} from '~/constants/types';
import styles from './styles';

interface IProps {
  center?: HeaderSubComponent;
  navigation: NavigationScreenProp<any, any>;
  left?: HeaderSubComponent;
  primary: TabColors;
  right?: HeaderSubComponent;
  styles?: StyleProp<ViewStyle>;
  theme: ITheme;
}

export default class Header extends React.Component<IProps, {}> {
  public render() {
    const {primary, theme} = this.props;
    return (
      <View>
        <StatusBar
          backgroundColor={theme.background}
          hidden={false}
          barStyle={theme.isDark ? 'light-content' : 'dark-content'}
        />
        <ElementsHeader
          centerComponent={this.props.center}
          containerStyle={[
            styles.container,
            {
              backgroundColor: theme.background as string,
              borderBottomColor: primary,
            },
            this.props.styles,
          ]}
          leftComponent={this.props.left}
          rightComponent={this.props.right}
        />
      </View>
    );
  }
}
