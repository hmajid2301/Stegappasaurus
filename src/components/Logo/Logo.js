import PropTypes from 'prop-types';
import React from 'react';
import { Image, Text, View } from 'react-native';

import styles from './styles';


const logo = require('../../assets/images/logo.png');


const Logo = ({ theme }) => (
  <View style={styles.container}>
    <Text style={[styles.text, { color: theme.color }]}>Steg</Text>
      <Image source={logo} style={styles.logo}/>
    <Text style={[styles.text, { color: theme.color }]}>ppasaurus</Text>
  </View>
);

Logo.propTypes = {
  theme: PropTypes.object.isRequired,
};

export default Logo;
