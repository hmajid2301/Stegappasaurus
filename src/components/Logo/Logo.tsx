import * as React from 'react';
import {Image, Text, View} from 'react-native';

import logoDark from '~/assets/images/logo-dark.png';
import logoLight from '~/assets/images/logo-light.png';
import {ThemeColors} from '~/constants/types';
import styles from './styles';

interface IProps {
  isDark: boolean;
  color: ThemeColors;
}

const Logo = ({color, isDark}: IProps) => (
  <View style={styles.logoContainer}>
    <Text style={[styles.text, {color}]}>Steg</Text>
    <Image source={isDark ? logoLight : logoDark} style={styles.logo} />
    <Text style={[styles.text, {color}]}>ppasaurus</Text>
  </View>
);

export default Logo;
