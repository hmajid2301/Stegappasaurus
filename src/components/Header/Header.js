import PropTypes from 'prop-types';
import React from 'react';
import { Header, Icon } from 'react-native-elements';

import Logo from '../Logo';
import styles from './styles';


const CustomHeader = ({ navigation, color }) => (
  <Header
    leftComponent={<Icon name='menu' type='simple-line-icon' color={styles(color).icons.color} onPress={() => navigation.openDrawer()}/>}
    centerComponent={<Logo />}
    outerContainerStyles={styles(color).outerHeaderContainer}
    innerContainerStyles={styles(color).innerHeaderContainer}
  />
);

CustomHeader.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  color: PropTypes.string.isRequired,
};

export default CustomHeader;
