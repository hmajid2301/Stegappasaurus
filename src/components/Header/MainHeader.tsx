import React from 'react';
import {TouchableOpacity} from 'react-native';
import {Icon} from 'react-native-elements';
import {NavigationScreenProp} from 'react-navigation';

import Logo from '~/components/Logo';
import {ITheme, TabColors} from '~/constants/types';
import Header from './Header';

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  primary: TabColors;
  theme: ITheme;
}

export default class MainHeader extends React.Component<IProps, {}> {
  public render() {
    return (
      <Header
        {...this.props}
        center={this.centerComponent()}
        right={this.rightComponent()}
      />
    );
  }

  private centerComponent = () => (
    <Logo color={this.props.theme.color} isDark={this.props.theme.isDark} />
  );

  private rightComponent = () => {
    const isSettings = this.isSettings();
    if (isSettings) {
      return (
        <TouchableOpacity>
          <Icon
            color={this.props.theme.color}
            name="settings"
            type="feather"
            onPress={this.toSettings}
          />
        </TouchableOpacity>
      );
    }
    return (
      <TouchableOpacity>
        <Icon
          color={this.props.theme.color}
          name="home"
          type="font-awesome"
          onPress={this.toHome}
        />
      </TouchableOpacity>
    );
  };

  private toSettings = () => {
    this.props.navigation.navigate('Settings');
  };

  private toHome = () => {
    this.props.navigation.navigate('Main');
  };

  private isSettings() {
    const routeName = this.props.navigation.state.routeName;
    let isSettingsPage = false;

    if (!routeName.startsWith('Settings')) {
      isSettingsPage = true;
    }

    return isSettingsPage;
  }
}
