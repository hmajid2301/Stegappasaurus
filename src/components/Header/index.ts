import AppHeader from './AppHeader';
import MainHeader from './MainHeader';

export {AppHeader, MainHeader};
