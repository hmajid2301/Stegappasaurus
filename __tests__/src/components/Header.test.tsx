import {shallow} from 'enzyme';
import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {Icon} from 'react-native-elements';
import {NavigationScreenProp} from 'react-navigation';

import Header from '~/components/Header/Header';
import Logo from '~/components/Logo';

jest.mock('react-navigation');
const navigation: NavigationScreenProp<any, any> = {
  openDrawer: jest.fn(),
  closeDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  dismiss: jest.fn(),
  goBack: jest.fn(),
  navigate: jest.fn(),
  getParam: jest.fn(),
  setParams: jest.fn(),
  emit: jest.fn(),
  addListener: jest.fn(),
  isFocused: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  dispatch: jest.fn(),
  state: {
    routes: [
      {
        key: 'Encoding',
        isTransitioning: false,
        index: 1,
        routes: [
          {routeName: 'Main', key: 'id-1565732047195-0'},
          {
            params: {
              uri:
                'file:/data/user/0/com.stegappasaurus/cache/ImagePicker/a89493b5-e2a5-4546-80f3-4fb13d2461d8.png',
            },
            routeName: 'Message',
            key: 'id-1565732047195-2',
          },
        ],
        routeName: 'Encoding',
      },
      {
        key: 'Decoding',
        isTransitioning: false,
        index: 0,
        routes: [{routeName: 'Main', key: 'id-1565732047195-1'}],
        routeName: 'Decoding',
      },
    ],
    index: 0,
    isTransitioning: false,
    key: 'Home',
    routeName: 'Home',
  },
};

describe('Header: Match Snapshots', () => {
  test('Blue Tab: Dark Theme', () => {
    const component = shallow(
      <Header
        navigation={navigation}
        primary="#009CFF"
        theme={{background: '#17212D', color: '#FFF', isDark: false}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test('Orange Tab: Light Theme', () => {
    const component = shallow(
      <Header
        navigation={navigation}
        primary="#E88C0C"
        theme={{background: '#FFF', color: '#17212D', isDark: true}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test('w/ Right Component', () => {
    const component = shallow(
      <Header
        navigation={navigation}
        right={
          <View>
            <Text>Hello!</Text>
          </View>
        }
        primary="#E88C0C"
        theme={{background: '#FFF', color: '#17212D', isDark: true}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test('w/ Left Component', () => {
    const component = shallow(
      <Header
        navigation={navigation}
        left={
          <TouchableOpacity>
            <Icon
              color="black"
              name="arrow-back"
              type="material-icons"
              onPress={() => null}
            />
          </TouchableOpacity>
        }
        primary="#E88C0C"
        theme={{background: '#FFF', color: '#17212D', isDark: true}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test('w/ Center Component', () => {
    const component = shallow(
      <Header
        navigation={navigation}
        center={<Logo color="#17212D" isDark={false} />}
        primary="#E88C0C"
        theme={{background: '#FFF', color: '#17212D', isDark: true}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test('w/ Styles', () => {
    const component = shallow(
      <Header
        navigation={navigation}
        primary="#E88C0C"
        styles={{height: 50}}
        theme={{background: '#FFF', color: '#17212D', isDark: true}}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
