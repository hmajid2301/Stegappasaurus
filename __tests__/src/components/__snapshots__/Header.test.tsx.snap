// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: Match Snapshots Blue Tab: Dark Theme 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#17212D"
    barStyle="dark-content"
    hidden={false}
    showHideTransition="fade"
  />
  <ForwardRef(Themed.Header)
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#17212D",
          "borderBottomColor": "#009CFF",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Header: Match Snapshots Orange Tab: Light Theme 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#FFF"
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
  />
  <ForwardRef(Themed.Header)
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#FFF",
          "borderBottomColor": "#E88C0C",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Header: Match Snapshots w/ Center Component 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#FFF"
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
  />
  <ForwardRef(Themed.Header)
    centerComponent={
      <Logo
        color="#17212D"
        isDark={false}
      />
    }
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#FFF",
          "borderBottomColor": "#E88C0C",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Header: Match Snapshots w/ Left Component 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#FFF"
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
  />
  <ForwardRef(Themed.Header)
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#FFF",
          "borderBottomColor": "#E88C0C",
        },
        undefined,
      ]
    }
    leftComponent={
      <TouchableOpacity
        activeOpacity={0.2}
      >
        <Themed.Icon
          color="black"
          name="arrow-back"
          onPress={[Function]}
          type="material-icons"
        />
      </TouchableOpacity>
    }
  />
</View>
`;

exports[`Header: Match Snapshots w/ Right Component 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#FFF"
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
  />
  <ForwardRef(Themed.Header)
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#FFF",
          "borderBottomColor": "#E88C0C",
        },
        undefined,
      ]
    }
    rightComponent={
      <View>
        <Text>
          Hello!
        </Text>
      </View>
    }
  />
</View>
`;

exports[`Header: Match Snapshots w/ Styles 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#FFF"
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
  />
  <ForwardRef(Themed.Header)
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#FFF",
          "borderBottomColor": "#E88C0C",
        },
        Object {
          "height": 50,
        },
      ]
    }
  />
</View>
`;
